format/spec.md

```diff
@@ -266,7 +266,11 @@ The `initial-default` is set only when a field is added to an existing schema. T
 
 The `initial-default` and `write-default` produce SQL default value behavior, without rewriting data files. SQL default value behavior when a field is added handles all existing rows as though the rows were written with the new field's default value. Default value changes may only affect future records and all known fields are written into data files. Omitting a known field when writing a data file is never allowed. The write default for a field must be written if a field is not supplied to a write. If the write default for a required field is not set, the writer must fail.
 
-All columns of `unknown`, `geometry`, and `geography` types must default to null. Non-null values for `initial-default` or `write-default` are invalid.
+All columns of `unknown`, `variant`, `geometry`, and `geography` types must default to null. Non-null values for `initial-default` or `write-default` are invalid.
+
+Default values for the fields of a struct are tracked as `initial-default` and `write-default` at the field level. Default values for fields that are nested structs must not contain default values for the struct's fields (sub-fields). Sub-field defaults are tracked in sub-field's metadata. As a result, the default stored for a nested struct may be either null or a non-null struct with no field values. The actual default value is produced by setting each field default in a new struct.
```

Comment:
non-null struct with no field values => empty struct?


```
The actual default value is produced by setting each field default in a new struct.
```

 Do we mean "New structs are produced by applying/combining all sub-field defaults."

Reply:
I think "non-null struct with no field values" is better because "empty" could be confusing in the context of a type. What does it mean to be empty? A struct with no field values means that there are no field-level values.

Maybe? I think that's less clear:
* "New structs are produced" is confusing because only the effective default value is produced this way
* "applying" or "combining" is subject to interpretation, but "setting" each field from its default is specific

Reply:
I thought 'empty' meant a struct with no field values? When I read "with no field values" i'm a little lost because what does that mean that isn't empty?

"Effective default value is produced" sounds good to me, "actual" seems ambiguous.